testing/simapp/app.go

```diff
@@ -133,7 +136,8 @@ var (
 		evidence.AppModuleBasic{},
 		transfer.AppModuleBasic{},
 		ibcmock.AppModuleBasic{},
-		ica.AppModuleBasic{},
+		icacontroller.AppModuleBasic{},
+		icahost.AppModuleBasic{},
```

Comment:
What about having a single ICA AppModule but 2 ICA IBC modules? This should fix the registration issue as the codec registration is conflicting due to the AppModule using the same RegisterInterfaces function (I'm fairly certain). The genesis would become shared (I think this is fine)

Reply:
That may be a perfect solution to the problem! I can look into this and follow up with you if I find any questions.

Reply:
@AdityaSripal maybe this approach can be used for `ccv` also! :)